public/content/documentation/web/page-level/form.md

```diff
@@ -1,80 +1,603 @@
+   - Instructions: Presented before the form or inline, announced in order.
+   - Errors: Errors are announced when inputs are invalid.
 
 Full information: [https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form](https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form)
 
 ## Gherkin
 
 ### #a11y - Web Accessibility Acceptance Criteria
 
-How to test a header
+How to test a web form for accessibility
 
-GIVEN THAT I am on a page with a header landmark
+GIVEN THAT I am on a page with a web form
 
 1. Keyboard for mobile & desktop
-
-   - WHEN I use the tab key to enter the web browser window I SEE focus is strongly visually indicated on interactive components
+   - WHEN I use the tab key to move focus to a form field
+      - I SEE focus is visually indicated
+      - I SEE the focus moves in a logical order
 
 2. Desktop screenreader
-
-   - WHEN I use a desktop screenreader (NVDA, JAWS, VoiceOver) AND
-
-   - I use the tab key to enter the web browser window
-
-   - I HEAR It is discoverable with screenreader shortcuts as header/banner landmark
-
-   - I HEAR It typically contains the name and primary navigation of the website
+   - WHEN I use a desktop screenreader (NVDA, JAWS, VoiceOver) AND 
+   - I use the tab key to move focus to inputs inside the form
+     - I HEAR If multiple forms are present (Ex: Search, Sign in, Newsletter subscription), the form must have a name
+     - I HEAR It is discoverable with screenreader shortcuts as a form landmark along with its name
+     - I HEAR the screenreader switch from browse shortcuts to forms shortcuts
 
 3. Mobile screenreader
-
    - WHEN I use a mobile screenreader (Talkback, VoiceOver) AND
-
-   - I swipe to focusable elements in the header
-
-   - I HEAR It is discoverable with screenreader shortcuts as header/banner landmark
-
-   - I HEAR It typically contains the name and primary navigation of the website
-
+   - I swipe to focus on inputs inside the form
+     - I HEAR If multiple forms are present (Ex: Search, Sign in, Newsletter subscription), the form must have a name
+     - I HEAR It is discoverable with screenreader shortcuts as a form landmark along with its name
+     - I HEAR the screenreader switch from browse shortcuts to forms shortcuts
 
 Full information: [https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form](https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/page-level/form)
 
 ## Developer Notes
 
-### Name
-
-- Typically doesn’t have a name or description since there must be only one instance per page.
+### Do not auto focus inputs
+
+Automatically moving focus to an input field is very confusing for people using assistive technology.
+
+### Group inputs
+
+Use `fieldset` and `legend` to group related fields, such as:
+- Sign in
+- Shipping address
+- Payment information
+
+### Error handling
+
+- Individual inputs must have programmatically described errors read by the screen reader on focus.
```

Comment:
missing inline links for [programmatically described errors](https://www.magentaa11y.com/checklist-web/hint-help-error/) and [alert](https://www.magentaa11y.com/checklist-web/alert/)